tomcat9/src/main/java/com/linecorp/armeria/server/tomcat/TomcatService.java

```diff
@@ -492,13 +494,20 @@ private static void convertHeaders(HttpHeaders headers, MimeHeaders cHeaders) {
             final AsciiString k = e.getKey();
             final String v = e.getValue();
 
-            if (k.isEmpty() || k.byteAt(0) == ':') {
+            if (k.isEmpty()) {
                 continue;
             }
 
-            final MessageBytes cValue = cHeaders.addValue(k.array(), k.arrayOffset(), k.length());
-            final byte[] valueBytes = v.getBytes(StandardCharsets.US_ASCII);
-            cValue.setBytes(valueBytes, 0, valueBytes.length);
+            if (k.byteAt(0) != ':') {
+                final byte[] valueBytes = v.getBytes(StandardCharsets.US_ASCII);
+                cHeaders.addValue(k.array(), k.arrayOffset(), k.length())
+                        .setBytes(valueBytes, 0, valueBytes.length);
+            } else if (HttpHeaderNames.AUTHORITY.equals(k) && cHeaders.getValue("host") == null) {
```

Comment:
Isn't it better to check 'host' header using `headers.get(HttpHeaderNames.HOST)` because `cHeaders` is being mutated.